python/semantic_kernel/semantic_functions/chat_prompt_template.py

```diff
@@ -1,8 +1,11 @@
 # Copyright (c) Microsoft. All rights reserved.
 
+import asyncio
 from logging import Logger
-from typing import TYPE_CHECKING, Dict, List, Optional, Tuple
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
 
+from semantic_kernel.connectors.ai.open_ai.models.chat.chat_message import ChatMessage
+from semantic_kernel.connectors.ai.open_ai.models.chat.function_call import FunctionCall
```

Comment:
Same comment about dependencies here -- I think it might make the most sense to move both these models out of the connector so that core classes don't take dependency on OpenAI connector.

Reply:
Now referencing a generic chat message (that is not related to OpenAI, just role and content), this is inherited by a OpenAIChatMessage. and that is used in a OpenAIChatPromptTemplate class.